test/sql-luatest/seq_scan_test.lua

```diff
@@ -0,0 +1,177 @@
+local server = require('luatest.server')
+local t = require('luatest')
+local g = t.group()
+
+g.before_all(function()
+    g.server = server:new({alias = 'seq_scan'})
+    g.server:start()
+    g.server:exec(function()
+        box.execute([[CREATE TABLE t (i INT PRIMARY KEY, a INT);]])
+        box.execute([[INSERT INTO t VALUES (1, 1), (2, 2), (3, 3);]])
+        box.execute([[CREATE TABLE s (i INT PRIMARY KEY, a INT);]])
+        box.execute([[INSERT INTO s VALUES (1, 1), (2, 2), (3, 3);]])
+    end)
+end)
+
+g.after_all(function()
+    g.server:exec(function()
+        box.execute([[DROP TABLE t;]])
+        box.execute([[DROP TABLE s;]])
+    end)
+    g.server:stop()
+end)
+
+g.test_seq_scan_error = function()
+    g.server:exec(function()
+        box.execute([[SET SESSION "sql_seq_scan" = false;]])
+        local t = require('luatest')
+
+        local _, err = box.execute([[SELECT * FROM t;]])
+        t.assert_equals(err.message, "Scanning is not allowed for 'T'")
+
+        _, err = box.execute([[SELECT * FROM t WHERE i + 1 > 2;]])
+        t.assert_equals(err.message, "Scanning is not allowed for 'T'")
+
+        _, err = box.execute([[SELECT * FROM t WHERE a > 2;]])
+        t.assert_equals(err.message, "Scanning is not allowed for 'T'")
+    end)
+end
+
+g.test_seq_scan_success = function()
+    g.server:exec(function()
+        box.execute([[SET SESSION "sql_seq_scan" = false;]])
+        local t = require('luatest')
+
+        local res, err = box.execute([[SELECT max(i) FROM t;]])
+        t.assert(res ~= nil and err == nil)
+
+        res, err = box.execute([[SELECT min(i) FROM t;]])
+        t.assert(res ~= nil and err == nil)
+
+        res, err = box.execute([[SELECT * FROM t WHERE i = 2;]])
+        t.assert(res ~= nil and err == nil)
+
+        res, err = box.execute([[SELECT * FROM t WHERE i > 2;]])
+        t.assert(res ~= nil and err == nil)
+
+        res, err = box.execute([[SELECT * FROM t WHERE i < 2;]])
+        t.assert(res ~= nil and err == nil)
+    end)
+end
+
+g.test_seq_scan_keyword = function()
+    g.server:exec(function()
+        box.execute([[SET SESSION "sql_seq_scan" = false;]])
+        local t = require('luatest')
+
+        local res, err = box.execute([[SELECT * FROM SEQSCAN t;]])
+        t.assert(res ~= nil and err == nil)
+
+        res, err = box.execute([[SELECT * FROM SEQSCAN t WHERE i > 2;]])
+        t.assert(res ~= nil and err == nil)
+
+        res, err = box.execute([[SELECT * FROM SEQSCAN t WHERE i + 1 > 2;]])
+        t.assert(res ~= nil and err == nil)
+
+        res, err = box.execute([[SELECT * FROM SEQSCAN t WHERE a > 2;]])
+        t.assert(res ~= nil and err == nil)
+    end)
+end
+
+g.test_seq_scan_joins = function()
```

Comment:
From the joins test it seems that JOINs never work without SEQSCAN. Aren't there cases when JOINs do search, not scanning?

Reply:
Thank you for the review! There is. Added a few tests.